test/test_macros.jl

```diff
@@ -1785,4 +1785,121 @@ function test_variable_Bool_argument()
     return
 end
 
+function test_nonlinear_generator_init_sum()
+    model = Model()
+    @variable(model, x)
+    a = @NLexpression(model, sum(x for i in 1:1; init = 0))
+    @test string(a) == "subexpression[1]: +x"
+    a = @NLexpression(model, sum(x for i in 1:0; init = 0))
+    @test string(a) == "subexpression[2]: 0.0"
+    a = @NLexpression(model, sum(x for i in 1:0; init = 1))
+    @test string(a) == "subexpression[3]: 1.0"
+    y = 3
+    a = @NLexpression(model, sum(x for i in 1:0; init = y))
+    @test string(a) == "subexpression[4]: 3.0"
+    a = @NLexpression(model, sum(x for i in 1:0, j in 1:0; init = 1))
+    @test string(a) == "subexpression[5]: 1.0"
+    a = @NLexpression(model, sum(x for i in 1:2, j in 1:2; init = 3))
+    @test string(a) == "subexpression[6]: 3.0 + x + x + x + x"
```

Comment:
I've updated, so now it is this. I don't know which is better. My guess is that people would probably expect this behavior, even if the docstring says otherwise?

Reply:
If, according to the docs, `init` must be the additive identify, maybe we should raise an error otherwise.

Reply:
The annoying part is that `Base` currently works:
```julia
julia> sum(i for i in 1:3; init = 2)
8
```

Reply:
The undefined behavior is unfortunate. This behavior of `init` is what I would expect, however.

Reply:
So happy to merge this for now? We can always add a warning later if the undefined behavior in `Base` changes or is confusing.

Reply:
Yes, happy to merge. Thank for fixing.